src/ashell/file-utils.c

```diff
@@ -170,3 +170,38 @@ int fs_get_boot_cfg_filename(const char *timestamp, char *filename)
     fs_close_alloc(file);
     return -1;
 }
+
+char *read_file_alloc(const char *file_name, ssize_t *size)
+{
+    char *file_buf = NULL;
+    fs_file_t *fp = fs_open_alloc(file_name, "r");
+
+    if (fp == NULL)
+        return 0;
+
+    *size = fs_size(fp);
+    if (*size == 0) {
+        ERR_PRINT("[ERR] Empty file (%s)\n", file_name);
```

Comment:
Drop the [ERR] from all ERR_PRINT calls, ERR_PRINT already puts (ERROR) in the output.

However, we want to be careful about use of ERR_PRINT since these strings all take space in our image. The string "out of memory" is used elsewhere so please use that exact string for any out of memory cases... then there will be only one copy in the image. (Bonus points: we should #define that string as a macro somewhere for spelling consistency and use the macro everywhere.) We don't really need more detail because ERR_PRINT will tell them the filename and line number where it occurred.

If you think more detail is warranted, like the filename here, you can put that in DBG_PRINT.

Reply:
Created a macro here for the error and simplified it to just "Read file failed". I've added DBG_PRINT as well in case someone needs to know more, it can be handy to know which file is failing and why.